lib/constructs/interface.js

```diff
@@ -163,9 +122,98 @@ class Interface {
           }
         }
       }
+    };
+
+    const seen = new Set([this.name]);
+    for (const member of this.members(seen)) {
+      let key;
+      switch (member.type) {
+        case "operation":
+          key = member.static ? "staticOperations" : "operations";
+          if (member.name) {
+            if (!this[key].has(member.name)) {
+              this[key].set(member.name, new Operation(this.ctx, this, member));
+            } else {
+              this[key].get(member.name).idls.push(member);
```

Comment:
Is the order of these guaranteed? We're sometimes accessing idls[0]. How do we know that that's the correct one? What's in the others?

Reply:
The order is not guaranteed. Here are the specific cases in which `idls[0]` is accessed:

- `utils.isOnInstance(this.idls[0], this.interface.idl)`

   This checks for `[Unforgeable]` extended attribute on `this.idls[0]` and returns true if that is the case. The Web IDL [forbids](https://heycam.github.io/webidl/#Unforgeable) some overloads of an operation having that extended attribute and others of the same operation do not:

   > If it does appear on an operation, then it must appear on all operations with the same identifier on that interface.

   We do not enforce this requirement at this moment, and a TODO is included:

   https://github.com/jsdom/webidl2js/blob/398c67c78193068bd018c19daa271b4c4c073adc/lib/constructs/operation.js#L28

   But assuming the input IDL is well-formed, checking the first overload only should suffice.

- `this.idls[0].name || this.name`

   This is a somewhat hacky way of supporting named stringifiers. Basically in case of

   ```webidl
   stringifier DOMString opName();
               DOMString opName(DOMString input);
   ```

   we create two `Operation`s:

   1. one with `op.name` being `"opName"`, but `op.idls` having both overloads, and
   2. the other's `op.name` property manually overridden with `"toString"`, but `op.idls[0]` is still `"opName"`, and `op.idls.length` is always 1.

   For the first case, since all overloads have the same name, choosing which one doesn't matter. For the second case, since there is always going to be one overload, the first and only one it is.

   OTOH, for a more basic case of 

   ```webidl
   stringifier;
   ```

   we know there is only always going to be one overload.

- `this.idls[0].stringifier`

   See above. This can be true only if there is one item in `this.idls`.

I can add some comments and asserts in the code regarding the last two items if that makes it a bit clearer?

Reply:
I'm fine with a single comment explaining this here (or at the access points, whatever seems clearer).